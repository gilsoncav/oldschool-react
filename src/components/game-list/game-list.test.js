import GameList from './game-list';

jest.mock('shortid', () => ({
  ...jest.requireActual('shortid'),
  generate: jest
    .fn()
    .mockReturnValueOnce('id1')
    .mockReturnValueOnce('id2')
    .mockReturnValueOnce('id3'),
}));

describe('GameList component', () => {
  it('renders correctly when isLoading', () => {
    const snapshot = snapRender(
      <GameList
        isLoading
        gamesArray={[]}
        maxTotalGames={24}
        gameOnClick={() => {}}
      />,
    );
    expect(snapshot).toMatchSnapshot();
  });
  it('renders correctly when has games', () => {
    const snapshot = snapRender(
      <GameList
        isLoading={false}
        gamesArray={[
          {
            Name: 'Super Mario Kart',
            Overview: "The best kart game in the world, y' now",
            Platform: 'Super Nintendo Entertainment System',
            VideoURL: null,
          },
          {
            Name: 'Super Mario Kart',
            Overview: "The best kart game in the world, y' now",
            Platform: 'Super Nintendo Entertainment System',
            VideoURL: null,
          },
          {
            Name: 'Super Mario Kart',
            Overview: "The best kart game in the world, y' now",
            Platform: 'Super Nintendo Entertainment System',
            VideoURL: null,
          },
        ]}
        maxTotalGames={24}
        gameOnClick={() => {}}
      />,
    );
    expect(snapshot).toMatchSnapshot();
  });

  it("renders correctly when don't has games", () => {
    const snapshot = snapRender(
      <GameList
        isLoading={false}
        gamesArray={[]}
        maxTotalGames={24}
        gameOnClick={() => {}}
      />,
    );
    expect(snapshot).toMatchSnapshot();
  });
});
