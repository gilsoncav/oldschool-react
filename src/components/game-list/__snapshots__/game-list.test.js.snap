// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameList component renders correctly when don't has games 1`] = `
.c0 h1 {
  margin-top: 25px;
  text-align: center;
  color: #ff6347;
}

.c0 .ant-col {
  margin-top: 20px;
  margin-bottom: 20px;
  height: 470px;
  min-height: 470px;
  max-height: 470px;
}

<div
  className="c0"
>
  <h1>
    There are no games available, please try again later...
  </h1>
</div>
`;

exports[`GameList component renders correctly when has games 1`] = `
.c0 h1 {
  margin-top: 25px;
  text-align: center;
  color: #ff6347;
}

.c0 .ant-col {
  margin-top: 20px;
  margin-bottom: 20px;
  height: 470px;
  min-height: 470px;
  max-height: 470px;
}

<div
  className="c0"
>
  <h1>
    Showing a maximum of 
    24
     games...
  </h1>
  <div />
</div>
`;

exports[`GameList component renders correctly when isLoading 1`] = `
.c0 h1 {
  margin-top: 25px;
  text-align: center;
  color: #ff6347;
}

.c0 .ant-col {
  margin-top: 20px;
  margin-bottom: 20px;
  height: 470px;
  min-height: 470px;
  max-height: 470px;
}

<div
  className="c0"
>
  <h1>
    <div
      className="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text"
    >
      <span
        className="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          className="ant-spin-dot-item"
        />
        <i
          className="ant-spin-dot-item"
        />
        <i
          className="ant-spin-dot-item"
        />
        <i
          className="ant-spin-dot-item"
        />
      </span>
      <div
        className="ant-spin-text"
      >
        Loading games...
      </div>
    </div>
  </h1>
</div>
`;
